pkg/pillar/dpcmanager/dpc.go

```diff
@@ -299,8 +304,8 @@ func (m *DpcManager) compressDPCL() {
 			m.Log.Tracef("compressDPCL: Adding Current Index: i = %d, dpc: %+v",
 				i, dpc)
 		} else {
-			// Retain the lastresort. Delete everything else.
-			if dpc.Key == LastResortKey {
+			// Retain the lastresort if enabled. Delete everything else.
+			if dpc.Key == LastResortKey && m.enableLastResort {
```

Comment:
Maybe we should also add test case for compressDPCL and enabled/disabled last resort: https://github.com/lf-edge/eve/blob/master/pkg/pillar/dpcmanager/dpc_test.go#L379

Reply:
Done